DistributionUpdate/PackageUpdate/PackageInfoTools.g

```diff
@@ -336,23 +336,46 @@ UpdatePackageInfoFiles := function(pkgdir)
                 "\n  This is not allowed, so the info file will not be changed\n");
           has_error := true;
         fi;
-        if IsBound(infon.Date) then
-          date := infon.Date;
-          if not IsString(date) and Length(date) = 10 and date{[3,6]} = "//" and
-            ForAll( date{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) then
-            Print("  ERROR (", info.PackageName, "): the date ", date,
-                  "should be a string of the form `dd/mm/yyyy'\n" );
-            has_error := true;
-          else
-            date := List( SplitString( date, "/" ), Int);
-            if not date[1] in [1..31] and
-                   date[2] in [1..12] and
-                   date[3] > 1999 and # GAP 4 appeared in 1999
-                   date[1] > DaysInMonth( date[2], date[3] ) then
-              Print("  ERROR (", info.PackageName, "): the date ", date,
-                    " is invalid\n" );
+        if IsBound( infon.Date ) then
+          if IsString( infon.date ) and Length( infon.Date ) = 10 
+            and infon.Date{ [3,6] } = "//" 
+            and ForAll( infon.Date{ [1,2,4,5,7,8,9,10] }, IsDigitChar ) then
+            date := List( SplitString( infon.Date, "/" ), Int ); # here date=[dd,mm,yyyy]
+            # the format dd/mm/yyyy is ambigous and can be confused with mm/dd/yyyy 
+            # if it is clear from the date that the format is mm/dd/yyyy 
+            # print a message
+            if date[2] in [13..31] and date[1] in [1..12] then
```

Comment:
So  `13/13/2020` would be OK? :-). Also you drop the check for 1999 but did not adjust the message. But why not keep the check? I'd put it into a newer local helper function, say,
```gap
IsPlausibleDate := date -> date[1] in [1..31] and
	                   date[2] in [1..12] and
	                   date[3] > 1999 and # GAP 4 appeared in 1999
	                   date[1] > DaysInMonth( date[2], date[3] );
```
Then you can write later on  `if not (IsBound(date) and IsPlausibleDate(date)) then ...` which IMHO is much more readable.

Reply:
At this point actually `13/13/2020` would be okay as this would be caught a couple of lines below when the validity of the date is actually checked. This check is only supposed to catch a date where it is plausible that the days and months are switched. In `13/13/2020` its just no date but e.g. in `12/24/2020` it is likely that it is a valid date but that the format is `mm/dd/yyyy`. I should make the comment above clearer though and also it makes no sense to print the same message twice for `12/24/2020`. But I like the idea that the whole check is smart enough to tell the user in such a situation that they've probably just switched days and months.